test/object-store/results.cpp

```diff
@@ -5016,6 +5016,135 @@ TEST_CASE("results: public name declared", "[results]") {
+        obj.set(int_col, (i + 2) % 4);
+        obj.set(str_col, "hello");
+        auto sub = obj.create_and_set_linked_object(link_col);
+        sub.set("weight", 60. + i);
+        sub.set("height", 170. + i);
+    }
+    realm->commit_transaction();
+    Results r(realm, table);
+
+    SECTION("test no operator equals") {
+        auto object_results = r.find("int_col = 0");
+        CHECK(object_results.size() == 2);
+        object_results = r.find(R""""(
+            {
+                "int_col": 0
+            }
+        )"""");
+        CHECK(object_results.size() == 2);
+        object_results = r.find(R""""(
+            {
+                "int_col": 1
+            }
+        )"""");
+        CHECK(object_results.size() == 2);
+        object_results = r.find(R""""(
+            {
+                "str_col": "hello"
+            }
+        )"""");
+        CHECK(object_results.size() == 8);
+        object_results = r.find(R""""(
+            {
+                "str_col": "bye"
+            }
+        )"""");
+        CHECK(object_results.size() == 0);
+    }
+    SECTION("test int comparison operators") {
+        auto object_results = r.find(R""""(
+            {
+                "int_col": { "$eq": 1 }
+            }
+        )"""");
+        CHECK(object_results.size() == 2);
+        object_results = r.find(R""""(
+            {
+                "int_col": { "$gt": 2 }
+            }
+        )"""");
+        CHECK(object_results.size() == 2);
+        object_results = r.find(R""""(
+            {
+                "int_col": { "$gte": 2 }
+            }
+        )"""");
+        CHECK(object_results.size() == 4);
+        object_results = r.find(R""""(
+            {
+                "int_col": { "$lt": 2 }
+            }
+        )"""");
+        CHECK(object_results.size() == 4);
+        object_results = r.find(R""""(
+            {
+                "int_col": { "$lte": 2 }
+            }
+        )"""");
+        CHECK(object_results.size() == 6);
+        object_results = r.find(R""""(
+            {
+                "link_col.weight": { "$gt": 62 }
+            }
+        )"""");
+        CHECK(object_results.size() == 5);
+        object_results = r.find(R""""(
+            {
+                "int_col": { "$in": [2, 3] }
+            }
+        )"""");
+        CHECK(object_results.size() == 4);
+    }
+    SECTION("test logical AND operator") {
+        auto object_results = r.find(R""""(
+            {
+                "$and": [
+                    { "int_col": { "$gt": 2 } },
+                    { "str_col": { "$eq": "hello" } }
+                ]
+            }
+        )"""");
+        CHECK(object_results.size() == 2);
+        object_results = r.find(R""""(
+            {
+                "$and": [
+                    { "int_col": { "$gt": 2 } },
+                    { "str_col": { "$eq": "bye" } }
+                ]
+            }
+        )"""");
+        CHECK(object_results.size() == 0);
+    }
+}
```

Comment:
Could you test a few non-happy paths to make sure that a reasonably worded syntax error is thrown?
Suggestions: 
- `{ "non-existing-property": "something" }`
- `{ str_col: { $regex: "(?i)a(?-i)cme" } }` (non-supported query operator "$regex")
- `{ "mal-formed-bson" : }`